nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -143,6 +143,7 @@ def run(self, data=None, commit=None):
         # Loop over each of the circuit maintenance records
         # pylint: disable=too-many-nested-blocks
         counter: int = 0
+        maintenance_list = []
```

Comment:
making this list "global", for all sites, means that if a maintenance affects circuits in multiple sites, it's only taken into account on the first site.

Reply:
the name `maintenance_list` is not super meaningful...

Reply:
Updated the var name. 

Thinking about the logic, it's more that overlap has been checked for in general. Just a list of maintenances, not specific for a site.

Reply:
I did dive down in on this. If we move the `processed_circuit_maintenances` down one loop, this causes an issue. There are too many overlaps found.

Can we pair on this in my morning? Perhaps 8:30am ET? @chadell